test/150_connect_forget_test.sh

```diff
@@ -13,11 +17,14 @@ weave_on $HOST2 launch
 start_container $HOST1 $C1/24 --name=c1
 start_container $HOST2 $C2/24 --name=c2
 
+assert_raises "has_direct_peer $HOST2 $HOST1" 1
 assert_raises "exec_on $HOST1 c1 $PING $C2" 1
```

Comment:
I should have spotted this before... it's dangerous to use `assert_raises` with a non-0 status check, since the test can fail for a multitude of reasons, and not necessarily the specific condition you are interested in. Use a `!`, or, if not in a shell context, `sh -c \"! ..."`.